### PR TITLE
fix(grpc): add the num_channels field inside the tests

### PR DESCRIPTION
**Pull request by vincenzopalazzo**
This adds some clean-up code on the rust side and also includes a pre-build check test on the rust side.

Currently, we do not have a strong test suite for rust that can fall inside our own task, maybe we could include it inside our make file logic in the future?

Fixes https://github.com/ElementsProject/lightning/issues/5971

### CODE REVIEW

**Comment by endothermicdev** [comment]
I think #5964 already covered the deprecation warning.

**Comment by vincenzopalazzo** [comment]
Thanks! @endothermicdev I removed the commit!

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -52,6 +52,8 @@ jobs:
         run: make -j 4 check-gen-updated
       - name: Check docs
         run: make -j 4 check-doc
+      - name: Check safety rust code
+        run: cargo test --all
```
[the remark this answers is not in the document]
Comment (vincenzopalazzo) [reply .github/workflows/ci.yaml:56]: you are right! thanks!

**Review by endothermicdev — APPROVED** [review]
ACK 27b51e02af6f5b7efdb1e7552d0af66f3d601fac

**Pull request merged**